### ISSUE TITLE
Android app: No resuming of upload after wifi signal loss

### ISSUE THREAD

**Issue by ltog** [mapillary/mapillary_issues#2437]
### Basic information
* Platform: `Android`
* Version: `321` 

### Steps to reproduce behavior

 1. Connect your smartphone to the internet using WiFi
 2. Start uploading a sequence
 3. Move smartphone into an area of weak signal strength
4. Move back into area of strong signal strength

### Expected behavior

The app should resume uploading.

### Actual behavior

The app doesn't resume uploading without manual intervention.

### Additional information

IIRC in an older version of the app there was an option to automatically (re)connect and try to upload images if a WiFi. This option seems to be gone in version 321. :-( According to my faded memory it worked better then...

**Comment by jfburke3** [comment]
I can't say that I've noticed a problem resuming upload after a wifi disconnect/reconnect in any of the recent releases, but it's possible that I just didn't have any sequences to upload with version 3.21 (I'm on 3.24 now, and it works fine resuming uploads).

**Comment by ltog** [comment]
Actually I'm not sure anymore if the photos really aren't uploaded. It may just seem that the upload was stuck due to the non-current progress indicator.

**Comment by katrinhumal** [comment]
@ltog Sorry for getting to this so late but we've had a couple of updates since your reports - could you try and see if you still have issues? It really should be resuming automatically when connection resumes, so if it doesn't, we need to look into it.

**Comment by ltog** [comment]
I checked again with version 3.36-SNAPSHOT: The upload indeed gets stuck after some time. It crossed my mind that the reason may be that the phone switches from 2.4 GHz to 5 GHz wifi (or vice-versa). My internet router provides wifi on both frequencies with different SSIDs.

Is there any more information I can provide to help?

**Comment by katrinhumal** [comment]
Thanks for your helpfulness @ltog! @peterneubauer, could this switching cause the uploads not to resume?

**Comment by peterneubauer** [comment]
Hi there,
I just checked disabling and enabling wifi while uploading in the upcoming next release - it's working for me, the upload is resuming upon the new network connecting. I suggest you join the beta program and re-test or wait a couple of days for this to come out. Please report back!

**Comment by ltog** [comment]
@peterneubauer : How did you disable wifi? By cutting the power to your router/AP?

I did try to join the beta program but am redirected to the Play Store, which I neither want nor technically can use with my phone operating system. Is joining the beta program equivalent to installing the latest snapshot version from http://android-releases.mapillary.io/index.html ?

**Comment by peterneubauer** [comment]
Try
http://android-releases.mapillary.io/Mapillary_Android_3.40-SNAPSHOT1.apk

I simply disabled Wifi on the phone and enabled it again. However, I will
try moving out of range of the Wifi and check some more.

/peter

**Comment by ltog** [comment]
@peterneubauer : I just checked version 3.40-SNAPSHOT1: Usually the upload stopped after some hundred pictures. This time it looks better: I could upload nearly thousand pictures before I got the message:

    Error java.net.SocketException: recvfrom failed: ECONNRESET (Connection reset by peer) 984 of 1186 photos uploaded

IIRC the previous messages said that the hostname couldn't be resolved. I'm not sure if this message now is an improvement. I can do some more tests...

**Comment by peterneubauer** [comment]
Hi there,
With a retry, things should continue? Next step, we will add information
about pending retry to the notification, it's on the list if things to
add...

/Peter

**Comment by ltog** [comment]
Yes, I'm able to continue uploading but it's a manual process and therefore annoying.

I'm looking forward to the modifications!

**Comment by peterneubauer** [comment]
Yes,
We will get there soon!

/Peter

**Comment by jfburke3** [comment]
I just want to note that I haven't observed this problem, at least not in a way that I can identify it as this issue.

**Comment by peterneubauer** [comment]
_mapillary_issues_ is no longer an active repository.

Let’s continue discussing features in our forum! We will import this issue as topic to the [Mapillary Forum](http://forum.mapillary.com) in the coming weeks. We’ll notify here once it’s been done. See you there!
